src/composables/fungibleTokens.ts

```diff
@@ -190,6 +191,9 @@ export function useFungibleTokens() {
         assetContractId = contractCallData.token;
       }
     }
+    if (protocol === PROTOCOLS.ethereum) {
+      assetContractId = toEthChecksumAddress(assetContractId);
+    }
```

Comment:
I strongly suggest to avoid doing any protocol specific logic in common files. Maybe the asset contract ID could be converted on the service level (in the function that normalizes the API response into our interfaces)?

Reply:
I've created a new task for this #2725 as this requires us to change this in many places